pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1074,7 +1074,8 @@ def visit_call(self, node: nodes.Call) -> None:
         self._check_super_with_arguments(node)
         self._check_consider_using_generator(node)
         self._check_consider_using_with(node)
-        self._check_use_list_or_dict_literal(node)
+        self._check_use_list_literal(node)
+        self._check_use_dict_literal(node)
```

Comment:
The call to ``inferred = utils.safe_infer(node.func)`` is pretty costly, I think we should keep list/dict together for better performance.

Reply:
I don't think we will prevent any calls to `utils.safe_infer` in this case. The list-check `_check_use_list_literal` will _only_ do it if `node.as_string() == "list()"` and `_check_use_dict_literal` will _always_ do it if `isinstance(node.func, astroid.Name)`¹.

The only thing we could improve are the checks `_check_use_dict_literal`¹, right? Doing both in one method will not have any effect. Or will it?

¹: I added a check there now: `node.as_string().startswith("dict(")`

Reply:
> ¹: I added a check there now: node.as_string().startswith("dict(")

That "solution" sucks. I will do `node.func.name == 'dict'`.

Reply:
Of course, this won't work for the following code:

```py
foo = dict
new_dict = foo(asdf=1)
```

But I think we can live with it for performance reasons?

Reply:
> I don't think we will prevent any calls to utils.safe_infer in this case. The list-check _check_use_list_literal will only do it if node.as_string() == "list()"

Fair point, I missed that.

> Of course, this won't work for the following code:
> 
> foo = dict
> new_dict = foo(asdf=1)
> 
> But I think we can live with it for performance reasons?


Yes definitely. Let's not punish everyone with slower linter to be accurate for those who redefine dict. They don't deserve a linter anyway 😄